15-greenhouse-monitor/README.md

```diff
@@ -0,0 +1,381 @@
+
+
+### Sensor Names
+
+The app senses and reports these values:
+
+* `air_temp`: the air temperature in Celsius
+
+* `air_humidity`: the relative humidity as a percent
+
+* `air_pressure`: the air pressure in kPa
+
+* `soil_moisture`: an indication of how much moisture is in the soil. A higher value means more moisture, a lower value means less moisture.
+
+* `soil_temp`: the approximate soil temperature
+
+* `light_level`: an indication of the ambient light level. A higher value means more ambient light, a lower value means less ambient light.
+
+The sensor names are used both to report the values in monitoring and alert events, and as part of the environment variable names that you use to configure [alert thresholds](#alerts).
+
+### Monitoring Events
+
+The app sends a monitoring event to `greenhouse.qo` every `report_mins` minutes. The event contains
+the readings of all sensors.
+
+```json
+{
+      "air_humidity": 34.414062,
+      "air_pressure": 1016.3697,
+      "air_temp": 24.15,
+      "app": "nf15",
+      "light_level": 273,
+      "soil_moisture": 344,
+      "soil_temp": 26.489365
+}
+```
+
+When alert thresholds are configured, and one or more sensor readings is outside of the normal range, the event also includes `alert` and `alert_seq` properties to indicate the severity of the alert and the progress of the alert. See the section on [Alerts](#alerts) for details about these properties.
+
+### Configuration Updates and Errors
+
+When environment variables are changed, the app posts an event to `notify.qo`. The event contains details of which values changed and any configuration errors.
+
+For example, setting the environment variable `environment_update_mins` to 1 produces this notification, indicating that the environment variable changed from the default of 5 minutes to 1 minute.
+
+```json
+{
+    "updates": {
+        "environment_update_mins": {
+            "new_value": 1,
+            "old_value": 5
+        }
+    }
+}
+```
+
+Changing several variables, with some of them set incorrectly leads to both updates and errors being present:
+
+```json
+{
+    "errors": {
+        "environment_update_mins": {
+            "error": "not a valid whole positive number.",
+            "value": "1e"
+        },
+        "soil_temp_warning_high": {
+            "error": "Value is not a number.",
+            "value": "abc"
+        }
+    },
+    "updates": {
+        "light_level_normal_high": {
+            "new_value": 2000
+        },
+        "light_level_normal_low": {
+            "new_value": 20
+        }
+    }
+}
+```
+
+## Alerts
+
+In addition to reporting the sensor values, the app can send an alert should a particular sensor value be too high or too low. The app uses environment variables to configure alert ranges for each sensor. The alert ranges are:
```

Comment:
```suggestion
In addition to reporting the sensor values, the app can send an alert when a particular sensor value is too high or too low. The app uses environment variables to configure alert ranges for each sensor. The alert ranges are:
```

Reply:
'be' lol. Not sure I wrote that, I wasn't even wearing a pirate hat. ;-)